ql/src/java/org/apache/hadoop/hive/ql/optimizer/lineage/Generator.java

```diff
@@ -72,18 +133,11 @@ public Generator(Set<String> hooks) {
   @Override
   public ParseContext transform(ParseContext pctx) throws SemanticException {
 
-    if (hooks != null && hooks.contains(ATLAS_HOOK_CLASSNAME)) {
-      // Atlas would be interested in lineage information for insert,load,create etc.
-      if (!pctx.getQueryProperties().isCTAS()
-          && !pctx.getQueryProperties().isMaterializedView()
-          && pctx.getQueryProperties().isQuery()
-          && pctx.getCreateTable() == null
-          && pctx.getCreateViewDesc() == null
-          && (pctx.getLoadTableWork() == null || pctx.getLoadTableWork().isEmpty())) {
-        LOG.debug("Not evaluating lineage");
-        return pctx;
-      }
+    if (!statementFilter.test(pctx)) {
```

Comment:
I have a fear that this change will break existing configs: Previously, we said that if the Atlas HiveHook is set, we will run that transformation. Now we are saying that if `hive.lineage.statement.filter` is set, we do the transformation.

Reply:
Yes, it breaks existing configs but previously Atlas HiveHook-related filters were hardcoded and impossible to test. Now `hive.lineage.statement.filter` is independent of hooks and hence applies for any hook making it possible to test using `LineageLogger`.